caikit/runtime/utils/servicer_util.py

```diff
@@ -238,13 +238,43 @@ def build_caikit_library_request_dict(
             "We are looping though these fields: %s",
             [field.name for field in request.DESCRIPTOR.fields],
         )
+        # get all oneof fields excluding from optional oneofs
+        one_of_fields = []
+        for oneof_name, oneof in request.DESCRIPTOR.oneofs_by_name.items():
```

Comment:
I'm probably missing something here, but I would expect that `request` here is a `DataObjectBase` and not a `proto`? The code says I'm wrong, but if I weren't, this logic to find non-optional oneofs would already be handled there and wouldn't need to be duplicated.

Reply:
From what I understand `request` is a proto since it's the argument that's directly passed from the `def Train(self, request, *_, **__)` within `caikit/runtime/servicers/global_train_servicer.py`.

Reply:
Ah, that makes sense